examples/todomvc/src/element.rs

```diff
@@ -0,0 +1,235 @@
+use crate::dbg;
+extern crate wasm_bindgen;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
+
+/// Wrapper for `web_sys::Element` to simplify calling different interfaces
+pub struct Element {
+    el: Option<web_sys::Element>,
+}
+
+impl From<Element> for Option<web_sys::Node> {
+    fn from(obj: Element) -> Option<web_sys::Node> {
+        if let Some(el) = obj.el {
+            Some(el.into())
+        } else {
+            None
+        }
+    }
+}
+
+impl From<Element> for Option<web_sys::EventTarget> {
+    fn from(obj: Element) -> Option<web_sys::EventTarget> {
+        if let Some(el) = obj.el {
+            Some(el.into())
+        } else {
+            None
+        }
+    }
+}
+
+impl Element {
+    pub fn qs(selector: &str) -> Option<Element> {
+        let body: web_sys::Element = web_sys::window()?.document()?.body()?.into();
+        let el = body.query_selector(selector).ok()?;
+        Some(Element { el })
+    }
+
+    /// Add event listener to this node
+    pub fn add_event_listener<T>(&mut self, event_name: &str, handler: T)
+    where
+        // TODO rewrite without static
+        T: 'static + FnMut(web_sys::Event) -> (),
+    {
+        let cb = Closure::wrap(Box::new(handler) as Box<FnMut(_)>);
+        if let Some(el) = self.el.take() {
+            let el_et: web_sys::EventTarget = el.into();
+            el_et.add_event_listener_with_callback(event_name, cb.as_ref().unchecked_ref());
+            cb.forget(); // TODO cycle collect this
+            if let Ok(el) = wasm_bindgen::JsCast::dyn_into(el_et): Result<web_sys::Element, _> {
+                self.el = Some(el);
+            }
+        }
+    }
+
+    // TODO fix static lifetimes
+    /// Delegate an event to a selector
+    pub fn delegate<T>(
+        &mut self,
+        selector: &'static str,
+        event: &str,
+        mut handler: T,
+        use_capture: bool,
+    ) where
+        T: 'static + FnMut(web_sys::Event) -> (),
+    {
+        if let Some(el) = self.el.take() {
```

Comment:
Could this use an early return to avoid the indentation?

```rust
let el = match self.el.take() {
    Some(e) => e,
    None => return,
};
// ...
```